src/main/java/org/commonjava/util/partyline/FileTree.java

```diff
@@ -327,46 +327,55 @@ boolean tryLock( File file, String ownerName, String label, LockLevel lockLevel,
         while ( end < 1 || System.currentTimeMillis() < end )
         {
             T result = tryLock( realFile, Thread.currentThread().getName(), "Open File for " + ( doOutput ? "output" : "input" ),
-                                           doOutput ? LockLevel.write : read, WAIT_TIMEOUT, unit, ( opLock ) -> {
+                                           doOutput ? LockLevel.write : read, timeout, unit, ( opLock ) -> {
                         FileEntry entry = entryMap.get( realFile.getAbsolutePath() );
-                        if ( entry.file != null )
+                        boolean proceed = false;
+                        synchronized ( entry )
```

Comment:
The entry in entryMap is used in many places. It is hard to understand if adding sync here but not elsewhere. In fact, there are many "lock"s and I get lost... e.g., there is a "operationLocks" which is a weak hashmap. The lock object will get collected after GC. And withOpLock() method creates a new lock for the operation to use. I don't know why. Isn't a lock need to be consistent? Is there any race condition if one operation is using the lock to do something, then other operation get a new lock and doing something on same target file?